sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -139,6 +141,36 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
   object JoinSelection extends Strategy
     with PredicateHelper
     with JoinSelectionHelper {
+    private val hintErrorHandler = conf.hintErrorHandler
+
+    private def checkHintBuildSide(
+        onlyLookingAtHint: Boolean,
+        buildSide: Option[BuildSide],
+        joinType: JoinType,
+        hint: JoinHint,
+        isBroadcast: Boolean): Unit = {
+      if (onlyLookingAtHint && buildSide.isEmpty) {
+        if ((isBroadcast && hintToBroadcastLeft(hint)) ||
```

Comment:
the code looks a bit messy, we can clearly show the branches:
```
def invalidBuildSideInHint(buildSide: String) = {
  hintErrorHandler.joinHintNotSupported(hint.leftHint.get,
    s"build $buildSide for ${joinType.sql.toLowerCase(Locale.ROOT)} join")
}
if (onlyLookingAtHint && buildSide.isEmpty) {
  if (broadcast) {
    // check broadcast hash join
    if (hintToBroadcastLeft) invalidBuildSideInHint("left")
    if (hintToBroadcastRight) invalidBuildSideInHint("right")
  } else {
    // check shuffle hash join
    ...
  }
}
```

Reply:
looks better